api/database/factories/PoolFactory.php

```diff
@@ -162,4 +163,17 @@ public function candidatesAvailableInSearch()
             ];
         });
     }
+
+    /** Add assessment steps to the pool
+     *
+     * @return \Illuminate\Database\Eloquent\Factories\Factory
+     */
+    public function withAssessments()
+    {
+        return $this->afterCreating(function (Pool $pool) {
+            AssessmentStep::factory()
+                ->count(3)
+                ->create(['pool_id' => $pool->id]);
```

Comment:
I'm unsure as we've got a pattern going with any model capable of creating its dependencies if needed be it pool, pool candidate, screening question, experiences, and so on.

Reply:
@petertgiles I'm interested in your new pattern here. What do you think is the biggest _disadvantage_ of allowing the child to generate a parent if the id isn't provided, instead of throwing the error?

Reply:
I'm still trying to figure out the best practice but I've found that that pattern can lead to chains where running a factory kicks off the creation of a bunch of unrelated models.  Also, the child doesn't necessarily know (or want to know) about the appropriate state to create the parent in so it makes sense to have the child exist.